lib/probes/tedious.js

```diff
@@ -5,22 +5,47 @@ const ao = require('..')
 const conf = ao.probes.tedious
 const Sanitizer = ao.addon.Sanitizer
 
+const logMissing = ao.makeLogMissing('tedious')
+
 module.exports = function (tedious) {
   const proto = tedious.Connection && tedious.Connection.prototype
-  if (proto) patchConnection(proto)
+  if (proto) {
+    patchConnection(proto)
+  } else {
+    logMissing('Connection.prototype')
+  }
   return tedious
 }
 
 function patchDefaultConfig (connection) {
-  if (typeof connection.defaultConfig !== 'function') return
-  shimmer.wrap(connection, 'defaultConfig', fn => function () {
+  // find a function that the constructor calls so that
+  // bindEmitter can be used on the 'this' returned by
+  // the constructor. this was originally defaultConfig()
+  // but that went away back in 2016.
+  const functions = ['createDebug', 'defaultConfig']
+  let f
+  for (let i = 0; i < functions.length; i++) {
+    if (typeof connection[functions[i]] === 'function') {
+      f = functions[i]
+      break
+    }
+  }
+  if (!f) {
+    logMissing('connection.[createDebug, defaultConfig]()')
+    return
+  }
+
+  shimmer.wrap(connection, f, fn => function () {
     ao.bindEmitter(this)
     return fn.apply(this, arguments)
```

Comment:
This looks magical ;) Where is `arguments` coming from?

Reply:
arguments is magical. it's the array-like object that is built in to JavaScript and contains all the arguments that were passed to a function (except arrow functions in which he still refers to the enclosing function's arguments).

so the arrow function `fn => function () {...}` is a function that accepts the argument `fn` and returns the function `function () {...}`. that function shows no arguments because it doesn't care (and because they might change). the only purpose of wrapping `f` is to bind `this` within the constructor. so `function() {...}` just binds `this` within `connection`'s constructor when it is calling `f`. then it goes ahead and calls `fn` normally. there are two forms of function calls in javascript when one wants to pass arbitrary arguments `fn.call()` and `fn.apply`. the first argument is the `this` to use. `call` enumerates each argument individually while `apply` has one argument which is an array (or array-like object) with all the arguments.

i'm not sure that makes it more clear or not. this was odd to me because it wasn't really caring about patching the function `fn`. it's only doing that so it can bindEmitter on `this`.

Reply:
Thanks, helps with understanding for sure. 
It's like with some foreign languages, I'm starting to understand but couldn't speak it.